contracts/supplementary/PayableExchangeIssue.sol

```diff
@@ -0,0 +1,234 @@
+
+        // exchange issue Base Set
+        exchangeIssueInstance.exchangeIssue(
+            _exchangeIssueData,
+            _orderData
+        );
+
+        address baseSetAddress = _exchangeIssueData.setAddress;
+        uint256 baseSetIssueQuantity = _exchangeIssueData.quantity;
+
+        // Approve base Set to transfer proxy
+        ERC20Wrapper.ensureAllowance(
+            baseSetAddress,
+            address(this),
+            transferProxy,
+            baseSetIssueQuantity
+        );
+
+        // Calculate the rebalancing Set issue Quantity
+        uint256 rebalancingSetIssueQuantity = calculateRebalancingSetIssueQuantity(
+            _rebalancingSetAddress,
+            baseSetIssueQuantity
+        );
+
+        // issue rebalancing set to the caller
+        coreInstance.issueTo(
+            msg.sender,
+            _rebalancingSetAddress,
+            rebalancingSetIssueQuantity
+        );
+
+        returnExcessFunds(baseSetAddress);        
+    }
+
+    /**
+     * Any unused Wrapped Ether or base Set issued is returned to the caller.
+     *
+     * @param _baseSetAddress    The address of the base Set
+     */
+    function returnExcessFunds(
+        address _baseSetAddress
+    )
+        private
+    {
+        // Return any excess base Set to the user
+        uint256 leftoverBaseSet = ERC20Wrapper.balanceOf(
+            _baseSetAddress,
+            address(this)
+        );
+        if (leftoverBaseSet > 0) {
+            ERC20Wrapper.transfer(
+                _baseSetAddress,
+                msg.sender,
+                leftoverBaseSet
+            );
+        }
+
+        // unwrap any leftover WETH and send eth back to the user
+        uint256 leftoverEth = ERC20Wrapper.balanceOf(
+            weth,
+            address(this)
+        );
+        if (leftoverEth > 0) {
+            wethInstance.withdraw(leftoverEth);
+            msg.sender.transfer(leftoverEth);
+        }
+    }
+
+    /**
+     * Given the issue quantity of the base Set, calculates the maximum quantity of rebalancing Set
+     * issuable.
+     *
+     * @param _rebalancingSetAddress    The address of the rebalancing Set
+     * @param _baseSetIssueQuantity     The quantity issued of the base Set
+     * @return rbSetIssueQuantity      The quantity of rebalancing Set to issue
+     */
+    function calculateRebalancingSetIssueQuantity(
+        address _rebalancingSetAddress,
+        uint256 _baseSetIssueQuantity
+    )
+        private
+        returns (uint256)
+    {
+        uint256[] memory rbSetUnits = ISetToken(_rebalancingSetAddress).getUnits();
+        uint256 rbSetUnit = rbSetUnits[0];
+        uint256 rbSetNaturalUnit = ISetToken(_rebalancingSetAddress).naturalUnit();
+
+        // Ensure that the base Set quantity is a multiple of the rebalancing Set natural unit
+        uint256 rbSetNormalizedBaseSetQuantity = _baseSetIssueQuantity.div(rbSetNaturalUnit).mul(rbSetNaturalUnit);
```

Comment:
I'm unsure what this line of code is supposed to do. Is this flipped with the line below? Wouldn't you want to see how many rebalancing set tokens you can issue and then make sure it's a multiple of the rebalancing set?

Reply:
+1 Think its flipped